svelte/src/lib/components/FileExplorer/Node.svelte

```diff
@@ -1,295 +1,338 @@
-                };
-                appStateDispatch({ type: AppActionType.AddTab, tab });
-            }
-        }
-    };
-
-    let position = { x: 0, y: 0 };
-    const handleRightClick = (e: any) => {
-        if (selectable) {
-            position = { x: e.clientX, y: e.clientY };
-            showMenuAt.set(nodePath);
-            appStateDispatch({ type: AppActionType.SelectFile, nodePath: nodePath });
-        }
-    };
-
-    const closeMenu = () => {
-        showMenuAt.set('');
-    };
-
-    const handleMenuNew = async (isDir: boolean) => {
-        addMode = true;
-        typeToAdd = isDir ? FileType.Folder : FileType.File;
-    };
-
-    const handleMenuRename = () => {
-        appStateDispatch({
-            type: AppActionType.UpdateFile,
-            nodePath: nodePath,
-            value: { editMode: true },
+        const tab: IGenericTab<IFileInfo> = {
+          id: nodePath,
+          title: label,
+          tooltip: fileSystemPath,
+          content: {
+            component: component,
+            data: {
+              folderPath: extractPath(fileSystemPath, uiContext.pathSeparator).parentPath,
+              fileName: label,
+            },
+          },
+        };
+        appStateDispatch({ type: AppActionType.AddTab, tab });
+      }
+    }
+  };
+
+  let position = { x: 0, y: 0 };
+  const handleRightClick = (e: any) => {
+    if (selectable) {
+      position = { x: e.clientX, y: e.clientY };
+      showMenuAt.set(nodePath);
+      appStateDispatch({ type: AppActionType.SelectFile, nodePath: nodePath });
+    }
+  };
+
+  const closeMenu = () => {
+    showMenuAt.set("");
+  };
+
+  const handleMenuNew = async (isDir: boolean) => {
+    addMode = true;
+    typeToAdd = isDir ? FileType.Folder : FileType.File;
+  };
+
+  const handleMenuRename = () => {
+    appStateDispatch({
+      type: AppActionType.UpdateFile,
+      nodePath: nodePath,
+      value: { editMode: true },
+    });
+  };
+
+  let relatedTestSuiteRelPaths: string[] = [];
+
+  const handleDeleteTestCase = async () => {
+    const meta = await codeGenerator.generateProjectMetadata($appState.projectFile!);
+    const testCase = meta?.testCases.find(
+      (tc) => combinePath([tc.folderPath, tc.fileName], uiContext.pathSeparator) === fileSystemPath
+    );
+
+    if (meta?.testSuites) {
+      relatedTestSuiteRelPaths = meta.testSuites
+        .filter((suiteFile) => suiteFile.content.testcases.some((tcId) => tcId === testCase?.content.id))
+        .map((suiteFile) => {
+          const suiteFileFullPath = combinePath([suiteFile.folderPath, suiteFile.fileName], uiContext.pathSeparator);
+          const suitesFolder = combinePath(
+            [$appState.projectFile!.folderPath, StandardFolder.TestSuites],
+            uiContext.pathSeparator
+          );
+          const suiteFileRelPath = suiteFileFullPath.replace(suitesFolder, "").substring(1);
+          return suiteFileRelPath.replaceAll(uiContext.pathSeparator, "/");
```

Comment:
Replace "/" by Node.ts->PATH_SEPARATOR  constant

However,  right here we have relatedSuiteFilePath  (absolute physical file path),  why do we parse to rel-path then pass into dialog,  but in the dialog, we parse from rel-path to  full-path again to modify test-suite file content?
Is it possible to pass the related-suites-full-path here into dialog?